env0/resource_environment.go

```diff
@@ -14,6 +14,47 @@ import (
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 )
 
+type SubEnvironment struct {
+	Id            string
+	Name          string
+	Revision      string
+	Configuration client.ConfigurationChanges `tfschema:"-"`
+}
+
+func getSubEnvironments(d *schema.ResourceData) ([]SubEnvironment, error) {
+	isubEnvironments, ok := d.GetOk("sub_environment_configuration")
+	if !ok {
+		return nil, nil
+	}
+
+	numberOfSubEnvironments := len(isubEnvironments.([]interface{}))
+
+	var subEnvironments []SubEnvironment
+
+	for i := 0; i < numberOfSubEnvironments; i++ {
+		prefix := fmt.Sprintf("sub_environment_configuration.%d", i)
+
+		var subEnvironment SubEnvironment
+
+		if err := readResourceDataEx(prefix, &subEnvironment, d); err != nil {
+			return nil, err
+		}
+
+		configurationPrefix := prefix + ".configuration"
+		if configuration, ok := d.GetOk(configurationPrefix); ok {
+			subEnvironment.Configuration = getConfigurationVariablesFromSchema(configuration.([]interface{}))
```

Comment:
Configuration is not handled by readResourceData (for various reasons). Luckily there's already a function that does it for us.